util/Logging.py

```diff
@@ -1,8 +1,72 @@
 # !SKA#0001 24/10/2022
 
 import datetime
-import discord
+import functools
 import re
+import sqlite3
+
+import discord
+
+from util.EmbedBuilder import EmbedBuilder
+
+
+def limit(_limit_level: int) -> callable:
+    """
+    Limit levels:
+    None: No limit
+    1: Cannot use database commands (I.e alerts & applications)
+    2: Cannot use pax non help specific commands (I.e. chemsearch, define, wiki, ...)
+    3: Cannot use pax. (ping, rule, tip...)
+    Levels are inclusive, so if you are level 2, you cannot use level 1 commands either.
+    """
+
+    def decorator(func: callable) -> callable:
+        @functools.wraps(func)
+        async def wrapper(*args, **kwargs) -> callable:
+            """
+            Wrapper to determine if the user is limited or not.
+            """
+
+            # Get the author id
+            author_id = None
+            for arg in args:
+                try:
+                    author_id = arg.author.id
+                    ctx = arg
+                except AttributeError:  # Type is not a context
+                    continue
+
+            if (
+                author_id is not None
+            ):  # if no author exists, then we cannot limit, return func instead
+                conn = sqlite3.connect("util/database.sqlite")
+                c = conn.cursor()
+                limit_level = (
+                    c.execute(
+                        "SELECT limitLevel from user where uid = ?",
+                        (author_id,),
+                    ).fetchone()[0]
+                    or 0
+                )  # if no limit level exists, then it is 0
+                if limit_level >= _limit_level:
+                    embed = EmbedBuilder(
+                        title="You cannot use this command!",
+                        description="You are limited from using this command! Please contact a moderator if you believe this is a mistake.",
+                        color=0xFF0000,
+                    ).build()
+                    await ctx.respond(embed=embed, ephemeral=True)
+                    Log(
+                        f"$ tried to use {ctx.command.name}. But is limited from using it.",
+                        ctx.author,
+                    )
+                    return
```

Comment:
Do not implicitly `return None` in function able to return non-`None` value
```suggestion
                    return None
```